src/s2/util/bits/bits.h

```diff
@@ -58,6 +58,10 @@
 #if defined(_MSC_VER)
 #include <intrin.h>
 #endif
+// keep compatibility with abseil-cpp LTS branch "lts_2020_09_23"
+#if !defined(ABSL_ASSUME)
+#define ABSL_ASSUME(cond) ABSL_INTERNAL_ASSUME(cond)
```

Comment:
Using the same definition as [here](https://github.com/abseil/abseil-cpp/blame/5ed77665c4d697f657fa1362b5a482450f858cdc/absl/base/optimization.h#L219), which should provide the same behaviour for code older than the definition of `ABSL_ASSUME `

Reply:
It makes me a bit uneasy to mess with `ABSL_` symbols in a header. How about something like

```
// TODO: Remove this and just use ABSL_ASSUME when it is in an LTS release.
#if defined(ABSL_ASSUME)
#define S2_ASSUME(cond) ABSL_ASSUME(cond)
#elif defined(ABSL_INTERNAL_ASSUME)
#define #S2_ASSUME ...
#else
#error "abseil-cpp must provide ABSL_ASSUME or ABSL_INTERNAL_ASSUME, what version are you using?"
#endif
```
?

Reply:
I understand your point, I was just taking a bit of code directly from the original project to make sure I copied the intended behaviour

To be clear, this suggestion would also involve replacing all calls to `ABSL_INTERNAL_ASSUME` with `S2_ASSUME` and eventually replace those with `ABSL_ASSUME ` right?

Reply:
Yes, thanks!